zebra/kernel_netlink.c

```diff
@@ -1039,7 +1039,6 @@ int netlink_request(struct nlsock *nl, struct nlmsghdr *n)
 	}
 
 	/* Fill common fields for all requests. */
-	n->nlmsg_flags = NLM_F_ROOT | NLM_F_MATCH | NLM_F_REQUEST;
```

Comment:
In this case, an individual MAC and IP is requested from  kernel. Kernel expect not to pass dump flag (NLM_F_ROOT | NLM_F_MATCH = NLM_F_DUMP). 
For specific MAC or IP only NLM_F_REQUEST flag needs to be passed. 
for dump of (MACS or IPs) within VLAN/SVI  NLM_F_DUMP flag is required. 
Based on that utilitizing same netlink request, simply moving and setting flags for NLM request to outer API whoever calls netlink_request. 

As other information is set for NLM from outer API for NLM so setting same flags from there.

Are you saying this netlink_request treated for dump purpose?

Reply:
I'm saying every single netlink request to the kernel is adding these flags and you removed this for everything.  I intentionally moved this into this function a while ago because everyone was using these flags.  Without some serious testing I'm more than a bit leary to make this change especially in the middle of *other* changes.  Break these commits up

Reply:
netlink_request() was only used for a table dump so far, but nothing in the API name signifies that. For requesting for a specific entry, these flags should not be set. I had thought about this earlier during my first look at the changes and didn't see anything wrong with the invokers of netlink_request() specifying the flags of interest to them - whether they want a table dump or not. A separate API can be created, but doesn't seem necessary.

Reply:
netlink_request() was created via 1fdc9eae2d80ec361e2a22d588944013a5d9b0b7 where the flags were set within the API.  I think by the api name (netlink_request), it is ok to use for the different netlink request made (i.e. can be dump or individual mac/ip request). There is no point creating separate api where doing the same logic. 
I will create two commits, one where flags are moved out of the netlink request and caller passes based on need. 
Second, where vxlan request specific MAC or IP via netlink request by setting required flag.